docs/rfc/active/react-app/README.md

```diff
@@ -0,0 +1,158 @@
+- Updating metadata entities  
+- Browsing metadata entities
+- Searching metadata entities
+- Managing a user account
+
+The finer details of which entities fall into each feature bucket will be dictated by the needs of the community, with the short-term milestone to achieve parity with entities appearing in the Ember client (Datasets, CorpUsers).
+
+
+#### Long term
+
+In the longer term, we will work with the community to define a more extensive functional road map, which may include
+
+- Providing migration pathway from the Ember application to the React application
+- New entities, aspects, operations (eg. Dashboards, Charts, etc)
+- Custom, server-driven ‘extensions’ or ‘applets’ to display in the UI
+- Admin Dashboard
+- Metrics Collection
+- Social features
+& more!
+
+### Architecture
+
+The figure below depicts the updated DataHub architecture given this proposal:
+
+![react-architecture](./react-architecture.png)
+
+Where the boxes outlined in green denote newly introduced components.
+
+Notice that the app will be completely independent of the existing Ember client, meaning there are no compatibility risks for existing deployments. Moreover, the React app will communicate exclusively with a GraphQL server (See [RFC 2042](https://github.com/linkedin/datahub/pulls?q=is%3Apr+is%3Aclosed) for proposal). This will improve the frontend development experience by providing 
+- a clearly defined API contract
+- simplified state management (via Apollo GQL client -- no redux required)
+- auto-generated models for queries and data types
+
+That’s the extent of the technical specifics we’ll cover for now. Stay tuned for a proof-of-concept PR coming soon that will present an initial React shell.
+
+## How we teach this
+
+A major goal of this initiative is to develop a frontend web client that can be easily extended by the DataHub community. Toward that end, we will provide documentation detailing the process of changing the frontend client to do things like:
+
+- Add a new entity page
+- Extend an existing entity page
+- Enable / disable specific features
+- Modify configurations
+- Test new components
+& more!
+
+
+## Alternatives
+
+### Evolve the Ember App in place
+
+*What?*: Iterate on the existing Ember appclient.
+
+*Why not?* Firstly, we actually do not consider this to be mutually exclusive with introducing a separate React app. Regardless, there are benefits to adopting a more accessible technology like React that do not change with improvements to the existing Ember app.
+
+### Mixing Ember & React
+
+*What?*: Migrate from Ember to React incrementally by incrementally replacing Ember components with React components.
+
+*Why not?*: The intermediate state of a half-react, half-ember app is something we’d rather not think about -- it’s scary & sad. We’d like to avoid degrading client-side developer experience with this type of complexity. Since this migration will take some time, we feel it more productive to iterate independently.
+
```

Comment:
In the interim ... Any new development happens in react / ember  / both?

- Developing in both can lead to poor developer experience.
- Developing in one can lead to drift.

Reply:
There are no great answers, once you commit to a dual world for a while. 

Here is a strawman that optimizes for autonomous parallel development: 
1) Get to React-Ember functional parity 
2) Recommend migration from Ember -> React
3) Ember feature freeze 
4) [eventually] deprecate Ember

Reply:
IMO this strategy (what flavor of dual development is sustainable) will emerge over time as well. We'll not lose sight of this as we iterate. Thanks @nagarjunakanamarlapudi!